pkg/auth/service.go

```diff
@@ -1902,13 +1903,44 @@ func NewAPIAuthService(apiEndpoint, token string, secretStore crypt.SecretStore,
 	} else {
 		cache = &DummyCache{}
 	}
-	res := &APIAuthService{
+	healthRes := &APIAuthService{
 		apiClient:   client,
 		secretStore: secretStore,
 		logger:      logger,
 		cache:       cache,
 	}
-	return res, nil
+
+	// Perform health check for API auth service
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = 5 * time.Second
+	bo.InitialInterval = 1 * time.Second
+	bo.MaxElapsedTime = 20 * time.Second
+	ctx := context.Background()
+	err = backoff.Retry(func() error {
+		healtResp, err := client.HealthCheck(ctx)
+		switch {
+		case err != nil:
+			return err
+		case healtResp.StatusCode == http.StatusNoContent:
+			return nil
+		default:
+			return fmt.Errorf("health check returned status %s: %w", healtResp.Status, ErrInvalidResponse)
+		}
+	}, bo)
+	if err != nil {
+		return nil, fmt.Errorf("health check failed: %w", err)
+	}
+
+	resp, err := client.GetVersionWithResponse(ctx)
```

Comment:
I'm not sure I would place version check  here, maybe keep it simple with Health check? 
Get version can be done externally if anyone wants. 
Healthcheck is already an indicator that the auth is alive.

Reply:
The version check was suggested by @nopcoder who pointed out that many services implement the health check endpoint. This will give us the certainty this is indeed the auth service who is responding

Reply:
@N-o-Z  Actually that's really smart I literally had an issue where I didn't know what was responding!